warp-core/src/main/scala/com/workday/warp/junit/WarpInfoExtension.scala

```diff
@@ -0,0 +1,49 @@
+package com.workday.warp.junit
+
+import java.util.stream.Stream
+
+import com.workday.warp.junit.TestIdConverters.extensionContextHasTestId
+import org.junit.jupiter.api.extension.{ExtensionContext, TestTemplateInvocationContext, TestTemplateInvocationContextProvider}
+import org.junit.platform.commons.util.AnnotationUtils
+
+
+/**
+  * A simple provider of invocation contexts that is solely meant to provide access to [[WarpInfo]], including testId.
+  *
+  * Note that we don't modify display name, or register a [[MeasurementExtension]] at all here.
+  *
+  * This extension should be used when your tests only need access to their names, and you want more fine-grained control
+  * over the ordering between extensions, eg if a [[MeasurementExtension]] before and after hooks would conflict with
+  * any other before/after hooks you may have.
+  *
+  * Created by tomas.mccandless on 10/23/20.
+  */
+trait WarpInfoExtensionLike extends TestTemplateInvocationContextProvider with TestIdSupport {
+
+  /**
+    * We only support test templates that are annotated with [[WarpInfoProvided]].
+    *
+    * @param context [[ExtensionContext]] containing test method and display name.
+    * @return whether the test method is annotated with [[WarpTest]].
+    */
+  override def supportsTestTemplate(context: ExtensionContext): Boolean = {
+    AnnotationUtils.isAnnotated(context.getTestMethod, classOf[WarpInfoProvided])
+  }
+
+
+  /**
+    * Provides a Stream of contexts corresponding to each invocation.
+    *
+    * In this case, we don't insert extra invocations, we only provide [[WarpInfo]] to a singular invocation.
+    *
+    * @param context [[ExtensionContext]] containing test method and display name.
+    * @return a singleton [[Stream]] of invocation context.
+    */
+  override def provideTestTemplateInvocationContexts(context: ExtensionContext): Stream[TestTemplateInvocationContext] = {
+    val testId: String = context.getTestId.get
+    val info: WarpInfo = WarpInfo(testId, 1, Warmup, 1, 0)
```

Comment:
Any reasoning behind why we want to label this singleton context as a  `Warmup` instead of `Trial`?

Reply:
good question, i was a little unsure what to do here. since `MeasurementExtension` isn't involved and the assumption is that measurement, if any, will take place as part of the test itself, rather than in one of our hooks, i felt `Warmup` fit better than `Trial`.

I could extend `RepetitionType` to define a 3rd type, or modify `WarpInfo` so that some of these fields are optional. I don't think I want to define another class similar to `WarpInfo`

what do you think?

Reply:
ah I see, hmm I like extending a 3rd type in `RepetitionType`, not sure about the naming though. `Passthrough`?

Reply:
yeah, tough to come up with a name for that. I could also move these particular fields to a different case class, like this:

```scala
case class RunInfo(currentRepetition: Int, repetitionType: RepetitionType, numWarmups: Int, numTrials: Int)
case class WarpInfo(testId: String, maybeRunInfo: Option[RunInfo] = None)
```

Reply:
up to you, could also just leave it as is and add a comment explaining the decision. I'm fine with any of those options

Reply:
cool, i ended up moving those fields to a different nested case class similar to the snippet from my prior comment